com.sap.cloud.lm.sl.cf.database/src/main/java/com/sap/cloud/lm/sl/cf/database/migration/executor/DatabaseMigrationExecutor.java

```diff
@@ -0,0 +1,34 @@
+package com.sap.cloud.lm.sl.cf.database.migration.executor;
+
+import java.sql.SQLException;
+import java.text.MessageFormat;
+
+import javax.sql.DataSource;
+
+import org.apache.log4j.Logger;
+
+import com.sap.cloud.lm.sl.cf.persistence.util.SqlQueryExecutor;
+
+public abstract class DatabaseMigrationExecutor {
+
+    private final static Logger LOGGER = Logger.getLogger(DatabaseMigrationExecutor.class);
+
+    public abstract DataSource getSourceDataSource();
+
+    public abstract DataSource getTargetDataSource();
+
+    public void executeMigration(String entityName) {
+        try {
+            executeMigrationInternal(entityName);
+        } catch (SQLException e) {
+            LOGGER.error(MessageFormat.format("Error migrating entity \"{0}\"", entityName), e);
+            throw new IllegalStateException(e.getMessage(), e);
```

Comment:
`IllegalStateException`s should be used only for things that should not happen. It's foreseeable for an error to occur here, so we should use some other exception type. Also, we should include `"Error migrating entity \"{0}\": {1}"` in the exception message, where 1 is the exception message of the cause.